src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -25,25 +26,35 @@
 import net.minecraftforge.common.capabilities.Capability;
 
 import javax.annotation.Nonnull;
-import java.util.regex.Pattern;
+import java.util.ArrayList;
+import java.util.List;
 
 
 public class CoverDetectorEnergyAdvanced extends CoverBehavior implements CoverWithUI, ITickable, IControllable {
```

Comment:
This should inherit from `CoverDetectorEnergy` and then you can clean up some of functions as base already handles it.

In similar way we were discussing on Advance ones for Fluid/Items

Reply:
resolved with https://github.com/GregTechCEu/GregTech/pull/1511/commits/40ab59e1e7f54457e22a4a5e3cbb923a8e7dd0b5